### PR TITLE
Cache summary data for observations, sample unit and sample events

### PR DESCRIPTION
**Pull request by gridcell**
Includes creating new set of models for the different sample units to store cached records from SQLModels for observation, sample unit and sample event summaries.

### CODE REVIEW

File: src/api/resources/sampleunitmethods/__init__.py [hunk 1]
```diff
@@ -166,6 +166,6 @@ def csv(self, request, *args, **kwargs):
 
     def get_queryset(self):
         project_id = self.kwargs.get("project_pk")
-        return self.model.objects.all().sql_table(
+        return self.model.objects.filter(
```
Comment (saanobhaai) [inline src/api/resources/sampleunitmethods/__init__.py:169]: This is my favorite diff line -- what a lot lies behind that small change...

File: src/api/utils/summary_cache.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+@timing
+def update_summary_cache(project_id, sample_unit=None, skip_test_project=True):
+    if (
+        skip_test_project is True
+        and Project.objects.filter(id=project_id, status=Project.TEST).exists()
+    ):
+        return
+
+    with transaction.atomic():
+        if sample_unit is None or sample_unit == FISHBELT_PROTOCOL:
```
Comment (saanobhaai) [inline src/api/utils/summary_cache.py:134]: Why is `if sample_unit is None` repeated for every if statement? Do we want to run `_update_cache` for every protocol if `sample_unit` is not specified?

**Review by saanobhaai — COMMENTED** [review]
This is a huge step forward...

File: src/api/utils/summary_cache.py (hunk 2 above)
Comment (gridcell) [reply src/api/utils/summary_cache.py:134]: if sample unit is None all the sample units need to run so I need to check it for each sample unit.  Unless I'm tired and missed something obvious (which could be the case)

**Pull request merged**